doc/updating_to_use_submodule.md

```diff
@@ -0,0 +1,10 @@
+# Updating to Submodule
```

Comment:
nit: rename file and heading as upgrading_to_v3.0.md

Reply:
Done.